### PR TITLE
Separator character between resource/extension

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
The naming module had hardcoded "_" for resource extensions

## Solution
Change the code to use the separator value from the naming module

## Tests
<Please provide steps to test the PR>

## Notes
<Additional comments for the PR>

### CODE REVIEW

**Review by rukawata — COMMENTED** [review]
Thanks for fixing this. Some problem showed up in pipeline - please check.

**Review by rukawata — COMMENTED** [review]
Seems that the separator is not passed:

> ~ name                = "U5584-EAUS-SAP-PRD_appSubnet-nsg" -> "U5584-EAUS-SAP-PRDappSubnet-nsg" # forces replacement
> ...
> ~ name                                           = "U5584-EAUS-SAP-PRD_app-subnet" -> "U5584-EAUS-SAP-PRDapp-subnet" # forces replacement

**Review by rukawata — COMMENTED** [review]
see error below. Thanks!

File: deploy/terraform/terraform-units/modules/sap_system/hdb_node/vm-hdb.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ HANA DB Linux Server private IP range: .10 -
 # Creates the admin traffic NIC and private IP address for database nodes
 resource "azurerm_network_interface" "nics_dbnodes_admin" {
   count = local.enable_deployment ? length(local.hdb_vms) : 0
-  name  = format("%s%s", local.hdb_vms[count.index].name, local.resource_suffixes.admin_nic)
+  name  = format("%s%s%s", local.hdb_vms[count.index].name, var.naming.separator, local.resource_suffixes.admin_nic)
```
[the remark this answers is not in the document]
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/hdb_node/vm-hdb.tf]: sorry - too eager copy paster

**Review by rukawata — COMMENTED** [review]
Found one more issue from pipeline :)
Currently can't test sap_landscape because terraform crashes during validation...

**Review by rukawata — DISMISSED** [review]
LGTM!

**Review by rukawata — APPROVED** [review]

**Pull request merged**